src/arch/traits.rs

```diff
@@ -13,6 +27,9 @@ use crate::internal::BeBytes;
 // TODO: add (optional?) trait methods for reading/writing specific register
 // (via it's GDB index)
 pub trait Registers: Default {
+    /// Register identifier for addressing single registers
+    type RegId: GdbSerializable + Default;
```

Comment:
Can't say I'm a fan of having a `Default` bound here. I understand why you've added it (from an implementation point of view), but there isn't really such thing as a "default register identifier" at the protocol level...

Reply:
+1, I'm happy you found a workaround !